backend/src/data/repositories/user/user.repository.ts

```diff
@@ -31,6 +38,45 @@ class User {
     return user ?? null;
   }
 
+  async getUserPermissons(id: number): Promise<PermissionM[] | null> {
+    const groups = await this.#UserModel
+      .query()
+      .select('groups.id')
+      .joinRelated('groups')
+      .where('users.id', id);
```

Comment:
About querying permissions through two many-to-many tables-resolvers - if you declared relation mappers correctly, then you could try something like that
```suggestion
    const groups = await this.#UserModel
      .query()
      .select('groups:permissions.id','groups:permissions.name', 'groups:permissions.key')
      .joinRelated('groups.permissions')
      .where('users.id', id);
```
resulting query looks like this
![image](https://user-images.githubusercontent.com/69236539/184013512-961e7f8f-c13b-4a9b-a973-cb0b92a9569d.png)
ps:: same query but readable
![image](https://user-images.githubusercontent.com/69236539/184013986-67eada00-fdab-4d3d-aa0e-56b66d972036.png)